code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,783 @@
+		if(7)
+			colour = "#FF00FF"
+
+/obj/effect/warped_rune/pyritespace/Crossed(atom/movable/AM, oldloc)
+	if(isliving(AM))
+		AM.add_atom_colour(colour, WASHABLE_COLOUR_PRIORITY)
+		activated = TRUE
+		playsound(src, 'sound/items/bikehorn.ogg', 50, TRUE)
+	. = ..()
+
+/obj/item/slimecross/warping/red
+	colour = "red"
+	runepath = /obj/effect/warped_rune/redspace
+	effect_desc = "Draw a rune that covers with blood whoever steps on it."
+
+/obj/effect/warped_rune/redspace
+	desc = "watch out for blood!"
+	icon_state = "rune_red"
+
+/obj/effect/warped_rune/redspace/Crossed(atom/movable/AM, oldloc)
+	if(ishuman(AM))
+		var/mob/living/carbon/human/H = AM
+		add_blood_DNA(list("Non-human DNA" = random_blood_type()))
+		for(var/obj/item/I in H.get_equipped_items(TRUE))
+			I.add_blood_DNA(return_blood_DNA())
+			I.update_icon()
+		for(var/obj/item/I in H.held_items)
+			I.add_blood_DNA(return_blood_DNA())
+			I.update_icon()
+		playsound(src, 'sound/effects/blobattack.ogg', 50, TRUE)
+		activated = TRUE
+	. = ..()
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	effect_desc = "Draw a rune that alters the DNA of those who step on it."
+	runepath = /obj/effect/warped_rune/greenspace
+
+/obj/effect/warped_rune/greenspace
+	desc = "Warning: don't step on this if you want to keep your genes."
+	icon_state = "rune_green"
+
+/obj/effect/warped_rune/greenspace/Crossed(atom/movable/AM, oldloc)
+	if(ishuman(AM))
+		randomize_human(AM)
+		activated = TRUE
+	. = ..()
+
+/* pink rune, makes people slightly happier after walking on it*/
+/obj/item/slimecross/warping/pink
+	colour = "pink"
+	effect_desc = "Draws a rune that makes people happier!"
+	runepath = /obj/effect/warped_rune/pinkspace
+
+/obj/effect/warped_rune/pinkspace
+	desc = "Love is the only reliable source of happiness we have left. But like everything, it comes with a price."
+	icon_state = "rune_pink"
+
+///adds the jolly mood effect along with hug sound effect.
+/obj/effect/warped_rune/pinkspace/Crossed(atom/movable/AM, oldloc)
+	if(istype(AM, /mob/living/carbon/human))
+		playsound(rune_turf, "sound/weapons/thudswoosh.ogg", 50, TRUE)
+		SEND_SIGNAL(AM, COMSIG_ADD_MOOD_EVENT,"jolly", /datum/mood_event/jolly)
+		to_chat(AM, "<span class='notice'>You feel happier.</span>")
+		activated = TRUE
+	. = ..()
+
+/obj/item/slimecross/warping/gold
+	colour = "gold"
+	runepath = /obj/effect/warped_rune/goldspace
+	effect_desc = "Draw a rune that exchanges objects of this dimension for objects of a parallel dimension."
+
+#define RARE_ITEM_LIST list(/mob/living/simple_animal/pet/dog/corgi/puppy/void,\
+							/obj/item/card/emagfake,\
+							/obj/item/clothing/head/speedwagon/cursed,\
+							/obj/structure/closet/crate/necropolis,\
+							/obj/item/gun/ballistic/revolver/reverse,\
+							/obj/item/grenade/gluon,\
+							/obj/item/sharpener,\
+							/obj/item/flashlight/flashdark,\
+							/mob/living/simple_animal/slime/rainbow,\
+							/obj/item/storage/belt/sabre,\
+							)
```

Comment:
You don't need a define here. Make a static list at the only proc using this define instead.